src/components/Drawer.svelte

```diff
@@ -0,0 +1,130 @@
+<div ref:wrapper class='coz-drawer-wrapper' on:click='set({folded: true})'>
+  <aside on:click='event.stopPropagation()'>
+    <hr class='coz-sep-flex' />
+    <nav>
+      {{#each groups as group}}
+      <NavigationGroup group='{{group}}' separator='top' />
+      {{/each}}
+    </nav>
+  </aside>
+</div>
+
+<script>
+  import NavigationGroup from './NavigationGroup'
+
+  let toggleDrawerObserver
+
+  export default {
+    data() {
+      return {
+        folded: true
+      }
+    },
+
+    onrender() {
+      toggleDrawerObserver = this.observe('folded', folded => {
+        if (folded) {
+          this.refs.wrapper.classList.toggle('visible', false)
+          setTimeout(() => {
+            this.refs.wrapper.setAttribute('aria-hidden', true)
+          }, 530)
+        } else {
+          this.refs.wrapper.setAttribute('aria-hidden', 'false')
+          setTimeout(() => {
+            this.refs.wrapper.classList.toggle('visible', true)
+          }, 30)
+        }
+      })
+    },
+
+    onteardown() {
+      toggleDrawerObserver.cancel()
+    },
+
+    components: {
+      NavigationGroup
+    }
+  }
+</script>
+
+<style>
+  .coz-drawer-wrapper {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+  }
+
+  .coz-drawer-wrapper[aria-hidden=true] {
+    display: none;
+  }
+
+  @media (min-width: 30.0625em) {
+    .coz-drawer-wrapper {
+      display: none;
+    }
+  }
+
+  .coz-drawer-wrapper::before {
+    content: '';
+    display: block;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background-color: var(--charcoal-grey);
```

Comment:
I noticed that you used custom properties (nice 😉) but aren't we supposed to support Edge browser? Because, afaik it doesn't support custom properties or not fully anyway